packages/core/src/connect.tsx

```diff
@@ -20,258 +23,208 @@ export const createUniverseWatcher = (universePath: string) => {
   return readProxy([universePath]);
 };
 
-const getValue = (path: string[], obj: any): any =>
-  path.reduce((x: any, y: any) => x && x[y], obj);
+const state = createUniverseWatcher("state");
 
-const valueExtractor = (
-  store: Store<any, any>,
-  watched: { [key: string]: any },
-) => (x: any) => {
-  const path = x[pathSymbol];
-  const pathKey = joinPath(path);
-  const value = getValue(path, store.universe);
-  watched[pathKey] = value;
-  return value;
+const useForceUpdate = () => {
+  const [, setCounter] = React.useState(0);
+  return React.useCallback(() => {
+    setCounter(tick => tick + 1);
+  }, []);
 };
 
-export const shallowEqual = (objA: any, objB: any): boolean => {
-  if (Object.is(objA, objB)) {
-    return true;
-  }
-
-  if (
-    typeof objA !== "object" ||
-    objA === null ||
-    typeof objB !== "object" ||
-    objB === null
-  ) {
-    return false;
-  }
-
-  const keysA = Object.keys(objA);
-  const keysB = Object.keys(objB);
-
-  if (keysA.length !== keysB.length) {
-    return false;
-  }
-
-  // Test for A's keys different from B.
-  // tslint:disable-next-line:prefer-for-of
-  for (let i = 0; i < keysA.length; i++) {
-    if (
-      !Object.prototype.hasOwnProperty.call(objB, keysA[i]) ||
-      !Object.is(objA[keysA[i]], objB[keysA[i]])
-    ) {
-      return false;
-    }
-  }
-
-  return true;
-};
+const getValue = (path: string[], obj: any): any =>
```

Comment:
This is the same as the `get` function in `watch.ts`. Maybe extract so its only defined once.